### PR TITLE
feat: Add UDT & common types to `avm/res/search/search-service` module

### PR DESCRIPTION
**Pull request by krbar**
## Description

Add UDT to `avm/res/search/search-service` module

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.res.search.search-service](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.search.search-service.yml)      |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/search/search-service/main.bicep [hunk 1]
```diff
@@ -91,11 +94,13 @@ param semanticSearch string?
 ])
 param sku string = 'standard'
 
+import { managedIdentityAllType } from 'br/public:avm/utl/types/avm-common-types:0.3.0'
 @description('Optional. The managed identity definition for this resource.')
-param managedIdentities managedIdentitiesType
+param managedIdentities managedIdentityAllType?
 
+import { diagnosticSettingFullType } from 'br/public:avm/utl/types/avm-common-types:0.2.1'
```
Comment (ReneHezser) [inline avm/res/search/search-service/main.bicep:101]: Any specific reason for not using the latest version here?

Comment (krbar) [reply avm/res/search/search-service/main.bicep]: @ReneHezser Thanks for pointing this out, no, just a copy/paste error. Fixed: https://github.com/Azure/bicep-registry-modules/pull/3767/commits/0f98fc429caddb6fd90d5489938fbc6f32ac4aa6

Comment (krbar) [reply avm/res/search/search-service/main.bicep]: @ReneHezser test successful.

**Review by ReneHezser — APPROVED** [review]
LGTM

**Pull request merged**